### PR TITLE
Add support for UI specified environment variables to application module

### PR DESCRIPTION
**Pull request by chrisghill**
We need to support environment variables specified in the UI.  This PR gets us there by adding capability for **all** of our app templates to have a top-level `envs` field that is a list of objects with `name` and `value` string fields.

This will merge the params envs (from the UI) with the app config envs (from `massdriver.yaml`).  If we like this, it might be a good idea to turn the `envs` block into a type so its very easy to add properly to any app template.

My thinking here is that all applications need to support the capability for users to specify environment variables (K8s, serverless containers and functions, VMs, etc) so its best to solve it for all of them at once.

### CODE REVIEW

File: massdriver-application/main.tf [hunk 1]
```diff
@@ -18,7 +19,10 @@ locals {
 
   policies = { for p in local.app_policies : p => jsondecode(data.jq_query.policies[p].result) }
 
-  base_envs = { for k, v in local.app_envs : k => jsondecode(data.jq_query.envs[k].result) }
+  base_envs = merge(
+    { for k, v in local.app_envs : k => jsondecode(data.jq_query.envs[k].result) },
+    { for env in local.params_envs: env.name => env.value }
```
Comment (chrisghill) [inline massdriver-application/main.tf:24]: `merge` in HCL handles collisions by prioritizing the last specified value, so here we are prioritizing params from the UI over those in the app definition.

File: massdriver-application/main.tf [hunk 2]
```diff
@@ -18,7 +19,10 @@ locals {
 
   policies = { for p in local.app_policies : p => jsondecode(data.jq_query.policies[p].result) }
 
-  base_envs = { for k, v in local.app_envs : k => jsondecode(data.jq_query.envs[k].result) }
+  base_envs = merge(
```
Comment (chrisghill) [inline massdriver-application/main.tf:22]: merge in HCL handles collisions by prioritizing the last specified value, so here we are prioritizing params from the UI over those in the app definition.

**Comment by WillBeebe** [comment]
It was my impression that we wanted to move away from having an array of envs. That if the user wanted an env, they would define a param per env. Then in the app block, specify any envs from the params.

**Comment by chrisghill** [comment]
I don't think we should burden users with having to learn JSON Schema and/or jq in order to do something as simple as setting an environment variable.

**Review by mclacore — APPROVED** [review]

**Review by coryodaniel — CHANGES_REQUESTED** [review]
Lets move any envs the user wants to expose on the params schema to a field that can be passed into the helm module `additionalEnv (list(object(name=string, value=string)) default: []`(?).

This will:
* allow us type the interface on the helm module
* not require a magic `envs` field in `massdriver-application` w/ magic subfields (name, value)
* push bundle devs towards using `app.envs` to document and validate their environment variables - we should add a note above params.envs

**Comment by chrisghill** [comment]
Closing this in favor of https://github.com/massdriver-cloud/terraform-modules/pull/49

**Pull request closed without merging**